packages/logger-plugin/src/action-logger.ts

```diff
@@ -22,14 +23,40 @@ export class ActionLogger {
   }
 
   completed(nextState: any) {
+    if (this.synchronousWorkEnded) {
+      const message = `(async work completed) ${this.getActionLogHeader()}`;
+      this.logWriter.startGroup(message);
+    }
     this.logWriter.logGreen('next state', nextState);
     this.logWriter.endGroup();
+    this.actionCompleted = true;
   }
 
   errored(error: any) {
+    if (this.synchronousWorkEnded) {
+      const message = `(async work error) ${this.getActionLogHeader()}`;
+      this.logWriter.startGroup(message);
+    }
     this.logWriter.logRedish('next state after error', this.store.snapshot());
     this.logWriter.logRedish('error', error);
     this.logWriter.endGroup();
+    this.actionCompleted = true;
+  }
+
+  syncWorkComplete() {
+    if (!this.actionCompleted) {
+      this.logWriter.logGreen('next state (synchronous)', this.store.snapshot());
+      this.logWriter.logGreen('( action doing async work... )', undefined);
+      this.logWriter.endGroup();
+    }
+    this.synchronousWorkEnded = true;
+  }
+
+  private getActionLogHeader() {
+    const actionName = getActionTypeFromInstance(this.action);
+    const formattedTime = formatTime(this.startedTime);
+    const message = `action ${actionName} (started @ ${formattedTime})`;
```

Comment:
is redunant, you can 

```ts
return `action ${actionName} (started @ ${formattedTime})`;
```

In addition, I recommend always describing the return type, since in the future you can forget to return an object instead of a returned string.

```ts
private getActionLogHeader(): string {
 
}
```

Reply:
Agreed. Will do